### PR TITLE
Support ray job 

### LINKED ISSUE #1120: Support ray job
Hello, I hope to take on the work of supporting rayjob in arena. Here is my general design：

Usage example:
```
arena submit rayjob \
    --name=rayjob-sample \
    --image=rayproject/ray:2.34.0 \
    --head-memory=4Gi \
    --worker-replicas=1  \
    --worker-memory=4Gi \
    --entrypoint="python /home/ray/samples/sample_code.py"
```

Parameters that need to be specified:

1. the name of ray job.
2. the image of ray job.
3. the command to be executed when submitting rayjob.
4. the cpu resource to HeadPod for the training.(default: 1)
5. the memory resource to HeadPod for the training.(default: 2Gi) 
6. the cpu resource to WorkerPod for the training.(default: 1)
7. the memory resource to WorkerPod for the training.(default: 1Gi)

### PR DESCRIPTION
**Pull request by qile123**
**What this PR does / why we need it**:
This PR is used to support ray job in arena.

**Which issue(s) this PR fixes** _(optional, in `Fixes #<issue number>, #<issue number>, ...` format, will close the issue(s) when PR gets merged)_:
Fixes #1120

### CODE REVIEW

**Comment by qile123** [comment]
@cheyang @Syulin7

**Comment by Syulin7** [comment]
@qile123  Thank you for this great contribution! Please rebase your PR.

**Comment by ChenYi015** [comment]
@qile123 Are you willing to contribute some docs about how to submit an example Ray job?

**Comment by qile123** [comment]
> Are you willing to contribute some docs about how to submit an example Ray job?

Certainly, I'm more than happy to contribute documentation on submitting a sample Ray job.

File: pkg/training/trainer_ray.go
Comment (ChenYi015) [inline pkg/training/trainer_ray.go:51]:
```suggestion
// RayJob implements the TrainingJob interface.
var _ TrainingJob = &RayJob{}
```

It would be better to add the above code to ensure that `RayJob` implements the `TrainingJob` interface.

File: pkg/training/trainer_ray.go
Comment (ChenYi015) [inline pkg/training/trainer_ray.go:234]:
```suggestion
// RayJobTrainer implements the Trainer interface.
var _ Trainer = &RayJobTrainer{}
```

File: pkg/training/trainer_ray.go
Comment (ChenYi015) [inline pkg/training/trainer_ray.go:321]: From my understanding, the chief pod of the ray job should be the job pod of the RayJob rather than the head pod of RayCluster, right?

The `rayJob` parameter is unused, do we need to check the job name retrieved from the pod labels and the ray job name are equal?

File: pkg/training/logs.go
Comment (ChenYi015) [inline pkg/training/logs.go:89]: As we discussed, arena will retrieve the latest chief pod by default, I think we do not need to add specific logic for ray job anymore.

**Comment by ChenYi015** [comment]
@qile123 Could you rebase this PR please? There are some merge conflicts due to the recent merged PRs.

**Comment by qile123** [comment]
> @qile123 Could you rebase this PR please? There are some merge conflicts due to the recent merged PRs.

sure

File: charts/rayjob/values.yaml
Comment (ChenYi015) [inline charts/rayjob/values.yaml:1]: I can see that some values used in the `rayjob.yaml` did not show up in this file. Would you like to update the `values.yaml` file to include all values used.

Comment (qile123) [reply charts/rayjob/values.yaml:1]: Ok, I will updata the values.yaml file.

**Review by ChenYi015 — APPROVED** [review]

**Comment by ChenYi015** [comment]
@qile123 Thanks for the great contribution of adding support for Ray job! LGTM, will wait for another approval @cheyang @Syulin7 .

**Comment by qile123** [comment]
> @qile123 Thanks for the great contribution of adding support for Ray job! LGTM, will wait for another approval @cheyang @Syulin7 .

Thank you for your positive feedback and for reviewing the changes that add support for Ray jobs! I appreciate your time and input.

**Comment by ChenYi015** [comment]
@qile123 Could you do a rebase, there is a merge conflict.

**Review by ChenYi015 — APPROVED** [review]

**Comment by cheyang** [comment]
Thanks for your contribution, @qile123

**Pull request merged**
